_data-prepper/pipelines/configuration/processors/flatten.md

```diff
@@ -0,0 +1,239 @@
+---
+layout: default
+title: flatten 
+parent: Processors
+grand_parent: Pipelines
+nav_order: 54
+---
+
+# flatten
+
+The `flatten` processor transforms nested objects inside of events into flattened structures. 
+
+## Configuration
+
+The following table describes configuration options for the `flatten` processor.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`source` | Yes | String | The source key to perform the operation on;. If set to an empty string (`""`), the processor uses the root of the event as the source.
+`target` | Yes | String | The target key to put into the flattened fields. If set to an empty string (`""`), the processor uses the root of the event as the target.
+`exclude_keys` | No | List | The keys from the source field that should be excluded from processing. Default is an empty list (`[]`).
+`remove_processed_fields` | No | Boolean | When `true`, the processor removes all processed fields from the source. Default is `false`.
+`remove_list_indices` | No | Boolean | When `true`, the processor converts the fields from the source map into lists and puts the lists into the target field. Default is `false`.
+`flatten_when` | No | String | A [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"'`, that determines whether the `flatten` processor will be run on the event. Default is `null`, which means all events will be processed unless otherwise stated.
+`tags_on_failure` | No | List | A list of tags to add to the event metadata when the event fails to process.
+
+## Usage
+
+The following examples show how the `flatten` processor can be used inside of Data Prepper pipelines.
+
+### Minimum configuration
+
+The following example shows only the required parameters to use the `flatten` processor, `source` and `target`:
```

Comment:
Do we mean something like "shows only the parameters that are required for using"?